File: server/tests/fixtures/alpha_rule_triggers/python/architecture_dependency_cycle_impact_partner.py
# architecture.dependency_cycle_impact fixture
# This file completes the impactful dependency cycle
# Should trigger: architecture.dependency_cycle_impact

from arch_dep_cycle_a import cycle_function_a

class CycleClassB:
    """Part of a cycle that affects architecture."""
    
    def call_a(self):
        return cycle_function_a()
    
def cycle_function_b():
    """Function in the cycle - creates circular dependency."""
    return CycleClassB()

File: server/tests/fixtures/alpha_rule_triggers/typescript/architecture_dependency_cycle_impact_partner.ts
// architecture.dependency_cycle_impact fixture
// This file completes the impactful dependency cycle
// Should trigger: architecture.dependency_cycle_impact

import { cycleFunctionA } from './arch_dep_cycle_a';

export class CycleClassB {
    callA() {
        return cycleFunctionA();
    }
}

export function cycleFunctionB() {
    return new CycleClassB();
}
